### PR TITLE
Time streaming exec scheduling

### LINKED ISSUE #42797: [Data] Measure overhead from scheduler (streaming executor) in `DatasetStats`
### Description

Measure the total time that the `StreamingExecutor` spends scheduling tasks.

### Use case

Additional stats collection for Ray Data performance debugging tool

### PR DESCRIPTION
**Pull request by omatthew98**
## Why are these changes needed?
Currently we are not timing how much time is spent during scheduling in the streaming executor. This times the total `process_time` for the scheduling steps / calls to `_scheduling_loop_step`. This stat is included in DatasetStats and a later PR will include this and other StreamingExecutor stats into the DatasetStatsSummary.

## Related issue number
Closes #42797 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Manual testing
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by omatthew98** [comment]
Conducted some manual testing with the following snippet (and adding a line to log the time from within `StreamingExecutor.run`):
```python
def sleep(x):
    time.sleep(0.5)
    return x

num_rows = sys.argv[1] if len(sys.argv) > 1 else 10

ds = ray.data.range(num_rows).map(sleep)

for _ in ds.iter_batches(batch_size=1):
    continue
```
For `num_rows = 100` the total scheduling time was `0.23742400000000297` and for `num_rows = 1000` the total scheduling time was `1.063609000000004`. More comprehensive testing will be added in followup pr which will update the metrics being reported out through `DatasetStatsSummary`.

**Review by scottjlee — APPROVED** [review]

File: python/ray/data/_internal/execution/streaming_executor.py [hunk 1]
```diff
@@ -208,8 +208,15 @@ def run(self):
...
-                pass
+            while True:
+                # use process_time to avoid timing ray.wait in _scheduling_loop_step
+                t_start = time.process_time()
+                continue_sched = self._scheduling_loop_step(self._topology)
+                self._initial_stats.streaming_exec_schedule_s.add(
+                    time.process_time() - t_start
+                )
+                if not continue_sched or self._shutdown:
+                    break
```
Comment (raulchen) [inline python/ray/data/_internal/execution/streaming_executor.py:219]: nit, better to use the context manager to make code cleaner and deal with exceptions

Comment (omatthew98) [reply python/ray/data/_internal/execution/streaming_executor.py:216]: Oh yeah much cleaner, thanks!

**Review by c21 — APPROVED** [review]

**Comment by can-anyscale** [comment]
Seem to broke linux://python/ray/data:test_streaming_integration

**Comment by can-anyscale** [comment]
I'm trying to revert to unblock test failures

**Pull request merged**